src/LayerPlanBuffer.cpp

```diff
@@ -94,10 +94,13 @@ void LayerPlanBuffer::addConnectingTravelMove(LayerPlan* prev_layer, const Layer
     if (!prev_layer->last_planned_position || *prev_layer->last_planned_position != first_location_new_layer)
     {
         const Settings& mesh_group_settings = Application::getInstance().current_slice->scene.current_mesh_group->settings;
-        const Settings& extruder_settings = Application::getInstance().current_slice->scene.extruders[prev_layer->extruder_plans.back().extruder_nr].settings;
+        const Settings& prev_extruder_settings = Application::getInstance().current_slice->scene.extruders[prev_layer->extruder_plans.back().extruder_nr].settings;
+        const Settings& next_extruder_settings = Application::getInstance().current_slice->scene.extruders[newest_layer->extruder_plans.back().extruder_nr].settings;
```

Comment:
If there are multiple extruders on `newest_layer`, then this will get the last of those extruders, not the first. Is this what you meant?

```suggestion
        const Settings& next_extruder_settings = Application::getInstance().current_slice->scene.extruders[newest_layer->extruder_plans.front().extruder_nr].settings;
```

Reply:
Yes, indeed. Good catch